plugins/myst/src/decode/inlines.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  CodeInline,
+  Emphasis,
+  Inline,
+  Strong,
+  Text,
+  Underline,
+} from "@stencila/types";
+import type { PhrasingContent } from "mdast";
+
+/**
+ * An interface for a MyST role https://mystmd.org/spec/overview#roles
+ *
+ * TODO: There is probably a published type for this amongst the MyST packages.
+ * (and for MyST directives). This is just an initial stab to get things type checking,
+ * look for published type, and if not available implement one based on above spec.
+ */
+interface MySTRole {
+  type: "mystRole";
+  name: string;
+  value: string;
+}
+
+type MySTInline = PhrasingContent | MySTRole;
+
+/**
+ * Transform MDAST `PhrasingContent` to Stencila Schema `Inline` nodes
+ *
+ * This is equivalent to the Rust `mds_to_inlines` function in
+ * `rust/codec-markdown/src/decode/inlines.rs`.
+ *
+ * This is also an update of code in
+ * https://github.com/stencila/encoda/blob/master/src/codecs/md/index.ts.
+ */
+export function mdsToInlines(mds: MySTInline[]): Inline[] {
+  return mds.map((md) => {
+    switch (md.type) {
+      case "text":
+        return new Text(md.value);
+      case "emphasis":
+        return new Emphasis(mdsToInlines(md.children));
+      case "strong":
+        return new Strong(mdsToInlines(md.children));
+      case "inlineCode":
+        return new CodeInline(md.value);
+      case "mystRole":
```

Comment:
Depending on how you are implementing the `mystRole`/`mystDirective` is mostly a parsing container, we generally get rid of these by the time we render / translate to other formats. For example here, we remove the `sup` role, and just use the children in a paragraph.

[![image](https://github.com/stencila/stencila/assets/913249/c30b0244-2e13-4545-9b17-6732f523ff2b)](https://mystmd.org/sandbox?tab=ast&myst=ewBzAHUAcAB9AGAAYQBzAGQAZgBgAA%3D%3D)

Reply:
Yep, I think we are on the same page here, this is just a `case` to unwrap the `mystRole`.

Reply:
There are a lot of transformations that we run on the AST, the most basic of which [are here](https://github.com/executablebooks/mystmd/blob/main/packages/myst-transforms/src/basic.ts#L18). This one [specifically lifts the roles/directives](https://github.com/executablebooks/mystmd/blob/feat/remote-url/packages/myst-transforms/src/basic.ts#L20).